lib/src/main/java/se/gustavkarlsson/track/sqlite/CloseableRecordCursorSequence.kt

```diff
@@ -8,15 +8,12 @@ internal class CloseableRecordCursorSequence(
 ) : CloseableSequence<Record> {
     private val sequence = sequence {
         cursor.use { cursor ->
-            var hasData = cursor.checkForDataOrClose()
-            while (hasData) {
-                val record = cursor.readExistingRecord()
-                hasData = cursor.checkForDataOrClose()
```

Comment:
I did it this way because of how [yield](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.sequences/-sequence-scope/yield.html) is documented to work.

The docs say it suspends until the next value is requested. I wanted it to always "preload" the next value so the cursor could close directly after loading the last value, and not after failing to load `last+1`

Does that make sense?

Either way, I might actually just remove this class completely and return a `List<Record>` instead. I don't like the idea that database cursors could be left open...

Reply:
I see. Regarding the open cursor, how about letting the user pass in a `Consumer<Sequence<Record>>` and then taking care of closing the cursor internally, taking away that responsibility from the calling code?

Reply:
This is one example where I had used this approach:
https://github.com/wowselim/java-imagehost/blob/master/imghost/src/main/java/co/selim/imghost/database/DBConnection.java#L28

Reply:
That's a really good idea! Maybe the signature for `query` should look something like this instead:

```kotlin
fun query(..., selector: (Sequence<Record>) -> List<Record> = { it.toList() }): List<Record>
```

**Pros:**
* By default, it just returns a list of all matching records.
* For advanced use, you can still save some performance by processing only part of the sequence.
* It would be "safe" in all cases

**Cons:**
* The `filters` lambda won't be the last argument any more, which makes that part a lot uglier :(